### ISSUE TITLE
[unbound-method] referencing a property with a function type should cause error

### ISSUE THREAD

**Issue by markfields** [typescript-eslint/typescript-eslint#1869]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/unbound-method": [
            "error",
            {
                "ignoreStatic": true
            }
        ]
  }
}
```

```TS
class SomeClass {
    functionProperty = function(this: any) { console.log(`this is ${this}`); };
}

const myFunctionProperty = new SomeClass().functionProperty;  // X - Missing eslint error
myFunctionProperty();  // Doesn't work, despite no eslint error
```

**Expected Result**
`unbound-method` error when initializing `myfunctionProperty `

**Actual Result**
No error, but calling `myFunctionProperty()` prints out `this is undefined`

**Additional Info**
I wonder if there should also be a rule for requiring the `this` parameter in inline function definitions be declared to have the most specific type possible(?).  If I declare `functionProperty = function(this: SomeClass) { ... }` then there's a compiler error when I try to call `myFunctionProperty()`:
```
The 'this' context of type 'void' is not assignable to method's 'this' of type 'MyClass'.ts(2684)
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.17.0` |
| `@typescript-eslint/parser`        | `2.17.0` |
| `TypeScript`                       | `3.7.5` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `10.19.0` |
| `npm`                              | `6.13.4` |

**Comment by bradzacher** [comment]
This happens due to an assumption that any property that is a function is an arrow function expression.
I don't think I've ever seen anyone use a non-arrow function expression in this way.

Happy to accept a PR to fix this.

Workaround is to use an arrow function instead.

**Comment by dinofx** [comment]
If you modify the example to:
`functionProperty = function(this: SomeClass)`

you get an error from typescript itself:
_The 'this' context of type 'void' is not assignable to method's 'this' of type 'SomeClass'._

**Comment by markfields** [comment]
@dinofx indeed - I noted the same in the original description.  I wonder if that would be a good linter rule too?
